src/Utilities/PredicateBuilder.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using ePiggyWeb.DataBase.Models;
+
+namespace ePiggyWeb.Utilities
+{
+    public static class PredicateBuilder
+    {
+        public static Expression<Func<IEntryModel, bool>> BuildEntryFilter(IEnumerable<int> idEnumerable, int userId)
+        {
+            if (idEnumerable is null)
+            {
+                return x => x.UserId == userId;
+            }
+
+            var idArray = idEnumerable as int[] ?? idEnumerable.ToArray();
+
+            if (!idArray.Any())
+            {
+                return x => x.UserId == userId;
+            }
+
+            Expression<Func<IEntryModel, bool>> filter = x => x.Id == idArray[0] && x.UserId == userId;
+
+            for (var i = 1; i <= idArray.Length; i++)
+            {
+                var i1 = i;
+                filter = filter.Or(x => x.Id == idArray[i1] && x.UserId == userId);
+            }
+
+            return filter;
+        }
```

Comment:
For example, what exactly this method does?

Reply:
(Lines 14 to 24 just check if the id list is not null or not empty, if it is just returns a predicate/filter by user id)
So let's say we have an array of ids, like [1,2,3] and user id = 0
We go by each id in the array, and we add up predicate/filter/lambda expression.
The predicate we would have at the end would look like: 

      x => (x.Id == 1 && x.UserId == 0) || (x.Id == 2 && x.UserId == 0) || (x.Id == 3 && x.UserId == 0)

I hope this explains it :D

Reply:
Oh, makes sense now. Well explained.